samples/styling-custom-visual/custom-visual/src/visual.ts

```diff
@@ -0,0 +1,138 @@
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+'use strict';
+
+import 'core-js/stable';
+import './../style/visual.less';
+import powerbi from 'powerbi-visuals-api';
+import VisualConstructorOptions = powerbi.extensibility.visual.VisualConstructorOptions;
+import VisualUpdateOptions = powerbi.extensibility.visual.VisualUpdateOptions;
+import IVisual = powerbi.extensibility.visual.IVisual;
+import EnumerateVisualObjectInstancesOptions = powerbi.EnumerateVisualObjectInstancesOptions;
+import VisualObjectInstance = powerbi.VisualObjectInstance;
+import DataView = powerbi.DataView;
+import VisualObjectInstanceEnumerationObject = powerbi.VisualObjectInstanceEnumerationObject;
+import IColorPalette = powerbi.extensibility.IColorPalette;
+import { TabulatorFormatter } from './tabulator-formatter';
+import { VisualSettings } from './settings';
+import Tabulator from 'tabulator-tables';
+
+export class Visual implements IVisual {
+    private target: HTMLElement;
+    private settings: VisualSettings;
+    private colorPalette: IColorPalette;
+    private errorMessageHTML: HTMLElement;
+    private dataViews : powerbi.DataView[]
+
+    constructor(options: VisualConstructorOptions) {
+        this.target = options.element;
+
+        if (document) {
+            this.target = document.createElement('div');
+            this.target.setAttribute('id', 'dynamicTable');
+
+            this.errorMessageHTML = document.createElement("p"); 
+            this.errorMessageHTML.style.display = "none";
+
+            this.colorPalette = options.host.colorPalette;
+            options.element.appendChild(this.target);
+            options.element.appendChild(this.errorMessageHTML);
+        }
+    }
+
+    public update(options: VisualUpdateOptions) {
+        this.settings = Visual.parseSettings(
+            options && options.dataViews && options.dataViews[0]
+        );
+        
+        /** Test 1: Data view has valid bare-minimum entries */
+        this.dataViews = options.dataViews;
+
+        if (
+            !this.dataViews ||
+            !this.dataViews[0] ||
+            !this.dataViews[0].table ||
+            !this.dataViews[0].table.rows ||
+            !this.dataViews[0].table.columns ||
+            !this.dataViews[0].metadata
+        ) {
+            this.errorMessageHTML.innerHTML = "Please drag an element into the values field";
+            this.target.style.display = "none";
+            this.errorMessageHTML.style.display = "block";
+            return;
+        }
+        
+        if(this.settings.dataTable.headerBackgroundColor === null){
```

Comment:
NIT: Add space after `if` and after `(`

Should look like this:

```
if (this.settings.dataTable.headerBackgroundColor === null) {
```

Please check if there are other places with similar issues.